python/hsfs/core/feature_view_engine.py

```diff
@@ -923,3 +935,138 @@ def _check_if_exists_with_prefix(self, f_name, f_set):
             )
         else:
             return f_name
+
+    def enable_feature_logging(self, fv):
+        self._feature_view_api.enable_feature_logging(fv.name, fv.version)
+        fv.enabled_logging = self.get(fv.name, fv.version).enabled_logging
+        return fv
+
+    def get_feature_logging(self, fv):
+        return FeatureLogging.from_response_json(
+            self._feature_view_api.get_feature_logging(fv.name, fv.version)
+        )
+
+    def _get_logging_fg(self, fv, transformed):
+        feature_logging = self.get_feature_logging(fv)
+        if transformed:
+            return feature_logging.transformed_features
+        else:
+            return feature_logging.untransformed_features
+
+    def log_features(self, fv, features, prediction=None, transformed=False, write_options=None, training_dataset_version=None, hsml_model=None):
+        default_write_options = {
+            "start_offline_materialization": False,
+        }
+        default_write_options.update(write_options)
+        fg = self._get_logging_fg(fv, transformed)
+        df = engine.get_instance().get_feature_logging_df(
+            fg,
+            features,
+            [feature for feature in fv.features if not feature.label],
+            [feature for feature in fv.features if feature.label],
+            FeatureViewEngine._LOG_TD_VERSION,
+            FeatureViewEngine._LOG_TIME,
+            FeatureViewEngine._HSML_MODEL,
+            prediction,
+            training_dataset_version,
+            hsml_model,
+        )
+        return fg.insert(df, write_options=default_write_options)
+
+    def read_log(self, fv,
+                 start_time: Optional[
+                     Union[str, int, datetime, datetime.date]] = None,
+                 end_time: Optional[
+                     Union[str, int, datetime, datetime.date]] = None,
+                 filter: Optional[Union[Filter, Logic]]=None,
+                 transformed: Optional[bool]=False,
+                 training_dataset_version=None,
+                 hsml_model=None,
+                 ):
+        fg = self._get_logging_fg(fv, transformed)
+        fv_feat_name_map = self._get_fv_feature_name_map(fv)
+        query = fg.select_all()
+        if start_time:
+            query = query.filter(fg.get_feature(FeatureViewEngine._LOG_TIME) >= start_time)
+        if end_time:
+            query = query.filter(fg.get_feature(FeatureViewEngine._LOG_TIME) <= end_time)
+        if training_dataset_version:
+            query = query.filter(fg.get_feature(FeatureViewEngine._LOG_TD_VERSION) == training_dataset_version)
+        if hsml_model:
+            query = query.filter(fg.get_feature(FeatureViewEngine._HSML_MODEL) == self.get_hsml_model_value(hsml_model))
+        if filter:
+            query = query.filter(self._convert_to_log_fg_filter(fg, fv, filter, fv_feat_name_map))
+        df = query.read()
+        df = df.drop(["log_id", FeatureViewEngine._LOG_TIME], axis=1)
+        return df
```

Comment:
Have you thought of removing this method and make the users get the features directly from the feature groups? They can read the features via `fv.feature_logging.transformed_feature_group.read()`.
It would simplify our API.

Reply:
I think it is better to hide the implementation detail as much as possible so that it is easier for us to make changes in the future. Also this method provide an easy way for user to filter log by time, td version and model version.

Reply:
Alright! that's fine.
What about renaming the method to `read_feature_logs`?

Reply:
renamed

Reply:
It was renamed in `feature_view_engine`, but not in `feature_view` :) Do you want to rename it there as well?

Reply:
Let change the name later after Jim come back